site/gatsby-site/src/utils/checklists.js

```diff
@@ -0,0 +1,121 @@
+import { classy } from 'utils/classy';
+import { faShield, faWarning } from '@fortawesome/free-solid-svg-icons';
+
+const abbreviatedTag = (tag) => tag.replace(/^.*:/g, '');
+
+const risksEqual = (risk1, risk2) => {
+  if (risk1.generated != risk2.generated) {
+    return false;
+  }
+  if (risk1.generated && risk2.generated) {
+    return risk1.title == risk2.title;
+  }
+  if (!risk1.generated && !risk1.generated) {
+    return risk1.id == risk2.id;
+  }
+};
+
+const Label = classy('label', 'mb-1 block');
+
+const emptyRisk = () => ({
+  id: Math.random().toString(36).slice(-10),
+  title: 'Untitled Risk',
+  tags: [],
+  risk_status: 'Not Mitigated',
```

Comment:
In the future, these strings and all status should come from a value library or equivalent.

Maybe a TODO, if this is prototyping? Things like this should block v1.

Reply:
I'm not familiar with the term "value library" and it seems hard to search, but I have switched to using a centralized object and manually throwing exceptions for non-existent keys in lieu of a static type system.

Reply:
Because of how JavaScript works, this doesn't do what anyone would expect it to do.

```javascript
const NOT_MITIGATED = 'Not Mitigated';
const PREVENTED = 'Prevented';

const riskStatusFeatures = {
    NOT_MITIGATED: {
        icon: faWarning,
        color: 'red',
    },
    PREVENTED: { ... },
}
```

`NOT_MITIGATED` is a valid unquoted object key, so using it here will result in:

```javascript
>>> riskStatusFeatures.NOT_MITIGATED
{ icon: faWarning, color: 'red' }

>>> riskStatusFeatures['Not Mitigated']
undefined

>>> riskStatusFeatures[NOT_MITIGATED]
undefined
```

I think there are ways to work around this, but it seems like treacherous territory.

Reply:
Goodness. I clearly didn't internalize an important edgy JS blog post on this topic. Let's run with what you got already then, thanks for exploring it.